Dining/forms.py

```diff
@@ -76,6 +86,15 @@ class Meta:
         model = DiningList
         fields = ['serve_time', 'min_diners', 'max_diners', 'sign_up_deadline', 'purchaser']
 
+    def clean(self, *args, **kwargs):
+        cleaned_data = super(DiningInfoForm, self).clean(*args, **kwargs)
+        # Check if the time is within time limits
+        serve_time = cleaned_data['serve_time']
+        if serve_time < settings.KITCHEN_USE_START_TIME:
+            self.add_error('serve_time', _("Kitchen can't be used this early"))
+        if serve_time > settings.KITCHEN_USE_END_TIME:
+            self.add_error('serve_time', _("Kitchen can't be used this late"))
```

Comment:
Needs `return cleaned_data`.

Also duplicate code, maybe better to put in model clean or MixIn or superclass.

Reply:
Used Mixin, that was a good idea.